Eastern-Research-Group/expert-query
# Feature/250 internal openapi spec

## Related Issues:
* [EQ-250](https://jira.epa.gov/browse/EQ-250)

## Main Changes:
* Created an OpenAPI spec for the private (non-CORS enabled) API endpoints.
* Fixed some mistakes in the public OpenAPI spec.
* Restructured the public OpenAPI spec, since there was odd behavior concerning some of the example values.
  * The request bodies for POST requests and the response schemas don't have example values now. At some point, we can add specific example objects, but it would entail more items to maintain, and I'm not sure if it's necessary since there are examples in the GET parameters and type schemas provided otherwise.
* Updated the Swagger API key.
* Updated the paths and descriptions to apply to Expert Query as a whole, rather than ATTAINS in particular.

## Steps To Test:
1. Review the public Swagger UI to ensure the changes match the actual routes.
2. Test the public API through Swagger. Currently, testing any server but `localhost` is not possible until the `api.data.gov` proxying is complete. Also, due to the React proxy, the port should be changed to 3000 when testing through the UI. I'm not sure if we should change the default value, since port 9090 should still be used when copying and pasting the requests.
3. Review the internal API spec. This is most easily done by changing `api-public.json` to `api-private.json` in line 153 of `app/server/app/routes/api.js`.

## TODO:
- [ ] We should add a top-level description and descriptions for individual parameters at some point, particularly when the user guide and glossary terms are drafted.
